### ISSUE TITLE
GNOME extension fails to appear in search/browse tab of Extension Manager

### ISSUE THREAD

**Issue by k0kubun** [xremap/xremap-gnome#4]
Transferred from https://github.com/k0kubun/xremap/issues/307, which was authored by @RedBearAK:

---

I have no idea what is causing this issue. I'm unfamiliar with the apps and systems involved with how GNOME extensions are submitted and listed as compatible with a particular version of GNOME. 

For some time now I've been testing a project of mine in a number of different Linux distros with GNOME desktops. The most convenient way to quickly find and install GNOME extensions for quite a while now has been the relatively new "Extension Manager" app that can be installed from Flathub. 

```sh
flatpak install extensionmanager
```

Or...

```sh
flatpak install com.mattjakeman.ExtensionManager
```

Several times now, I've installed that app on a distro I'm testing (mostly in virtual machines using GNOME Boxes/QEMU) and searched for the `Xremap` extension, only to see "No Results Found", even though the Xremap extension supports that version of GNOME. This has happened on different versions of GNOME. 

In this app, you can choose "Show Unsupported" from the hamburger menu icon in the header bar. By doing this, the Xremap extension will finally appear. Normally unsupported extensions will have a gray button with "Unsupported" instead of a blue "Install" button. 

When the Xremap extension finally appears after selecting "Show Unsupported" from the menu, it has a blue "Install" button and can be installed. So the problem is that it just won't appear in searches without using the "Show Unsupported" option. Which is pretty strange. 

My theory is that there is something wrong with the metadata for the Xremap extension that the Extension Manager app uses to determine if extensions are "supported" by that version of GNOME. Something must be causing a glitch that keeps the Xremap extension from being available in the Browse tab without that option in the menu. 

This is affecting me because I'm using the Xremap extension as one of three different GNOME extensions I've discovered that provide the application class and window name/title over D-Bus, to allow the `keyszer` keymapper to work in Wayland+GNOME. The other two are "Focused Window D-Bus" and "Window Calls Extended". With those two extensions I've never had this issue with them not appearing when searching in Extension Manager. In fact I don't know that I've seen this problem with any other GNOME extension, with the extension clearly being compatible with that version of GNOME, but not showing up unless I enable "Show Unsupported". 

The issue with Xremap not showing up while searching seems to mostly happen with the newer versions of GNOME, like GNOME 44. On the other hand, with older versions of GNOME like 3.38.x on Zorin OS, the Xremap extension is the only one of the three that is compatible and appears in search. So my project can work in a Wayland session on Zorin OS only because of the Xremap extension, but frequently in the newer versions of GNOME the Xremap extension will appear to not exist, at least when trying to search for it from within this application. 

Note that the app I'm talking about is not the same as the older "Extensions" app, which does not have any kind of browse or search ability, but was just a way to enable or disable extensions.

**Comment by k0kubun** [comment]
> This is affecting me because I'm using the Xremap extension as one of three different GNOME extensions I've discovered that provide the application class and window name/title over D-Bus, to allow the keyszer keymapper to work in Wayland+GNOME.

First off, I'd like to reiterate that the xremap-gnome extension exists only for xremap users. I shouldn't use my maintenance resources which are supported by GitHub Sponsors for something that I wasn't paid to maintain.

Therefore, can you please consider one of the following options?

1. Keep using xremap-gnome, but contribute missing features to xremap so that you can use xremap-gnome for xremap.
2. Create a `keyszer`-dedicated fork of this extension and switch to it.

I'm assuming you only use keyszer and don't use xremap at all, but I'm sorry if it's not the case.

**Comment by RedBearAK** [comment]
@k0kubun 

> I'm assuming you only use keyszer and don't use xremap at all, but I'm sorry if it's not the case.

I think I never got a notification about this comment. 

I don't have any interest in asking you to direct resources toward fixing the GNOME extension in any way specifically for my usage outside of `xremap`. I use the extension completely as-is, as part of a set of three extensions that do something similar, and if it changes I will adapt my project to its changes, or fork it as you suggested. 

When I submitted the original issue, I thought of this report as something that might also be affecting `xremap` users who were searching for the extension in the Extension Manager app, which has become a common, convenient way to install shell extensions. So I reported what seemed to be happening. 

At the time the searching bug seemed to be only affecting your extension, but it's been an issue on and off with other extensions since. With Extension Manager apparently being just a front end for the extensions site (the Browse tab stopped working when they had the big site outage recently), it probably means there was never anything different about the Xremap extension on the site. And therefore there would be nothing to fix. 

It follows that this issue should probably be closed.
